geolib/models/dsettlement/dsettlement_model.py

```diff
@@ -206,10 +206,17 @@ def set_any_calculation_options(self, **kwargs):
 
         calculation_options = self.datastructure.calculation_options.dict()
         calculation_options.update(**kwargs)
-        self.datastructure.calculation_options = CalculationOptions.set_options(
+        self.datastructure.calculation_options = CalculationOptions.set_imaginary_surface_options(
             **calculation_options
         )
 
+        """Check that the given layer index refers to an existing index layer"""
+        options = self.datastructure.calculation_options
+        if options.is_imaginary_surface \
+                and (options.imaginary_surface_layer < 1 or
+                     options.imaginary_surface_layer > self.layers.layers.__len__()):
+            raise ValueError("The index of imaginary_surface_layer refers to a not existing index layer.")
```

Comment:
instead of 'not existing' I think 'nonexisting' is better.
Did you really mean 'index layer' or just 'layer' or 'layer index'?